cloud/ec2_client.go

```diff
@@ -861,6 +861,17 @@ func (c *awsClientImpl) CreateFleet(ctx context.Context, input *ec2.CreateFleetI
 				}
 				return true, err
 			}
+			// CreateFleetWithContext has oddball behavior. If there is a
+			// RequestLimitExceeded error while using CreateFleet in instant mode,
+			// usually (but not always!) EC2.CreateFleetWithContext will not return an
+			// error. Instead it will populate the output.Error array with a single
+			// item describing the error, using an error type that does _not_
+			// implement awserr.Error. We therefore have to check this case in addition
+			// to the standard `err != nil` case above.
+			if len(output.Errors) > 0 {
+				grip.Error(message.WrapError(errors.New(output.Errors[0].String()), msg))
```

Comment:
I wouldn't worry about doing wrapError/errors new here, just log the string, and do a messageFields thing here with more data, or append the Errors output to that message (i'm also a little dubious about what other errors might be included in this array.

Reply:
Although admittedly awkward, the reason I prefer this is that it makes the log message have the same structure as all other log messages in this file, which is helpful when searching across them in Splunk, which I do often.